pilot/pkg/model/validation.go

```diff
@@ -1502,6 +1502,15 @@ func validateDestinationWeights(weights []*networking.DestinationWeight) (errs e
 		if weight.Destination == nil {
 			errs = multierror.Append(errs, errors.New("destination is required"))
 		}
+		for name := range weight.AppendHeaders {
+			errs = appendErrors(errs, ValidateHTTPHeaderName(name))
+		}
+		for _, name := range weight.RemoveResponseHeaders {
+			errs = appendErrors(errs, ValidateHTTPHeaderName(name))
+		}
+		for name := range weight.AppendResponseHeaders {
+			errs = appendErrors(errs, ValidateHTTPHeaderName(name))
+		}
```

Comment:
Also please add removerequestheaders. i think its okay to change the API to make it appendRequest/REsponseHeaders.. Envoy now supports both. https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/route/route.proto

Reply:
Is there an istio/api PR that goes along with this change? There were other name changes suggested in post-merge comments of https://github.com/istio/api/pull/626. Those should also be made at the same time.